Documentation/Evolution/RegexLiteralPitch.md

```diff
@@ -0,0 +1,230 @@
+
+As an example, the above regex may be transformed into:
+
+```swift
+let regex = {
+  var builder = T.RegexLiteral()
+  
+  // __A3 == /([0-9A-F]+\d)/
+  let __A1 = builder.customCharacterClass(["0"..."9", "A"..."F"])
+  let __A2 = builder.oneOrMore(__A1)
+  let __A3 = builder.characterClassDigit()
+  let __A4 = builder.concatenate(__A2, __A3)
+  let __A5 = builder.captureGroup(__A2)
+  
+  // __B4 == /\.\.([0-9A-F]+)/
+  let __B1 = builder.literal("..")
+  let __B2 = builder.customCharacterClass(["0"..."9", "A"..."F"])
+  let __B3 = builder.oneOrMore(__B2)
+  let __B3 = builder.captureGroup(__B3)
+  let __B4 = builder.concatenate(__B1, __B3)
+  
+  // __C3 == /__A5(?:__B4)?/
+  let __C1 = builder.group(__B4)
+  let __C2 = builder.zeroOrOne(__C1)
+  let __C3 = builder.concatenate(__A5, __C2)
+  
+  return T(regexLiteral: builder.finalize(__C3))
+}()
+```
+
+Note the exact API being used here is out of scope for this pitch.
+
+### The `ExpressibleByRegexLiteral` and `RegexLiteralProtocol` protocols
+
+New `ExpressibleByRegexLiteral` and `RegexLiteralProtocol` protocols will be introduced to the standard library, and will serve a similar purpose to the existing literal protocols `ExpressibleByStringInterpolation` and `StringInterpolationProtocol`.
+
+```swift
+public protocol ExpressibleByRegexLiteral {
+  associatedtype RegexLiteral : RegexLiteralProtocol = Regex
+  init(regexLiteral: RegexLiteral)
+}
+
+public protocol RegexLiteralProtocol {
+  init()
+  
+  // Informal builder requirements for building a regex literal
+  // will be specified here.
+}
+```
+
+Types conforming to `ExpressibleByRegexLiteral` will be able to provide a custom type that conforms to `RegexLiteralProtocol`, which will be used to build the resulting regex value. A default conforming type `Regex` will be provided by the standard library, the details of which are out of scope for this pitch.
+
+Giving conforming types the flexibility to provide their own regex handling logic allows for e.g additional linguistic-level processing to be built on top of regular expressions (e.g. using Unicode word-break iterators, collation and/or locale support, fuzzier matching, etc). It could also be used to provide regex types that call directly into `NSRegularExpression`, or other existing regex libraries.
+
+### Opting into certain regex features
+
+We intend for the compiler parsing logic to completely parse [the PCRE syntax][PCRE], however types conforming to `RegexLiteralProtocol` may not be able to handle the full regex grammar. To signal the features that they support, conforming types will be able to annotate builder functions with availability; which allows them to communicate the features they don't intend on supporting, in addition to the features they may support in the future.
+
+This also would give the standard library the flexibility to roll out new regex features over time.
+
+### Impact of using `/` as the delimiter
+
```

Comment:
I think it makes sense to start with the easy/benign and work forwards from there. So probably start with comment syntax, etc., which will also help the reader develop their syntactic reasoning.